### PR TITLE
releng: Remove 1.14 jobs, shift beta version marker, and regenerate tests

### PR DESCRIPTION
**Pull request by justaugustus**
Fixes: https://github.com/kubernetes/test-infra/issues/15514, fixes: https://github.com/kubernetes/kubernetes/issues/87111 (cc: @kubernetes/ci-signal)

- Remove Kubernetes 1.14 jobs
- Update generated tests for pre-1.18
- Update existing branch job version refs using config-rotator
- provider/gcp: Comment out gce-device-plugin-gpu-beta dashboard tab

For Release Engineering bits:
/assign @Katharine @tpepper @cpanato @saschagrunert

For kind:
/assign @BenTheElder @aojea 

cc: @kubernetes/release-engineering 
/priority critical-urgent

### CODE REVIEW

**Comment by justaugustus** [comment]
Ughhhh. I'm gaining an even deeper appreciation for the work @Katharine did to get us to the point we are at today with the release branches and job generation.

I think I know what this next failure is about. Time to poke at the config-rotator.

**Comment by neolit123** [comment]
LGTM on the `config/jobs/kubernetes/sig-cluster-lifecycle/*` changes.
i have a PR open for this here:
https://github.com/kubernetes/test-infra/pull/15576

but was waiting for 1.17.1 to come out.

**Comment by justaugustus** [comment]
This is about to fail again. Here's what I got locally:

```shell
$ time bazel test //config/...
INFO: Analyzed 3 targets (1 packages loaded, 981 targets configured).
INFO: Found 3 test targets...
FAIL: //config/tests/testgrids:go_default_test (see /home/augustus/.cache/bazel/_bazel_augustus/e12d25d9a9ef0205fe09fd32cbe17129/execroot/io_k8s_test_infra/bazel-out/k8-fastbuild/testlogs/config/tests/testgrids/go_default_test/test.log)
INFO: From Testing //config/tests/testgrids:go_default_test:
==================== Test output for //config/tests/testgrids:go_default_test:
--- FAIL: TestConfig (0.01s)
    config_test.go:228: Dashboard wg-resource-management, tab gce-device-plugin-gpu-1.12: - Testgroup ci-kubernetes-e2e-gce-device-plugin-gpu-beta must be defined first
    config_test.go:187: Dashboard sig-node-cri-1.12: - Must have more than one dashboardtab
    config_test.go:328: Testgroup ci-kubernetes-node-kubelet-features-beta - defined but not used in any dashboards.
    config_test.go:328: Testgroup ci-kubernetes-node-kubelet-beta - defined but not used in any dashboards.
    config_test.go:333: Note: Testgroups are automatically defined for postsubmits and periodics.
    config_test.go:334: Testgroups can be added to a dashboard either by using the `testgrid-dashboards` annotation on the prowjob, or by adding them to testgrid/config.yaml
--- FAIL: TestKubernetesProwInstanceJobsMustHaveMatchingTestgridEntries (1.90s)
    config_test.go:436: Testgrid group ci-kubernetes-node-kubelet-beta does not have a matching jenkins or prow job
    config_test.go:436: Testgrid group ci-kubernetes-node-kubelet-features-beta does not have a matching jenkins or prow job
FAIL
================================================================================
INFO: Elapsed time: 5.614s, Critical Path: 2.30s
INFO: 2 processes: 2 linux-sandbox.
INFO: Build completed, 1 test FAILED, 2 total actions
//config/tests/jobs:go_default_test                             (cached) PASSED in 3.1s
//config/tests/lint:go_default_test                             (cached) PASSED in 1.1s
//config/tests/testgrids:go_default_test                                 FAILED in 2.1s
  /home/augustus/.cache/bazel/_bazel_augustus/e12d25d9a9ef0205fe09fd32cbe17129/execroot/io_k8s_test_infra/bazel-out/k8-fastbuild/testlogs/config/tests/testgrids/go_default_test/test.log

Executed 1 out of 3 tests: 2 tests pass and 1 fails locally.
INFO: Build completed, 1 test FAILED, 2 total actions

real	0m5.650s
user	0m0.017s
sys	0m0.017s
```

Looks like there are some more old jobs to prune/reconfigure.

**Comment by justaugustus** [comment]
PR now includes a cleanup of the testgrid annotations for `node-kubelet` jobs, so I'm tagging SIG Node for awareness.

cc: @dchen1107 @derekwaynecarr

**Comment by alejandrox1** [comment]
@justaugustus why are 1.14 jobs being removed at this time?

**Comment by justaugustus** [comment]
Ahhh, the first merge conflict.
Rebased on master, which includes https://github.com/kubernetes/test-infra/pull/15882.

**Comment by justaugustus** [comment]
/hold as we're discussing potentially cutting another 1.14 release to support kubeadm skew jobs (cc: @neolit123)

[ref](https://github.com/kubernetes/kubernetes/issues/86182#issuecomment-573476889):
> kubeadm historically maintains `<OLDEST-VERSION-IN-SKEW>-1` upgrade and "skew" jobs.
> this allows us to catch regressions in `<OLDEST-VERSION-IN-SKEW>` for users who are moving slower with the upgrades.
> 
> in this case `<OLDEST-VERSION-IN-SKEW>-1 == 1.14`.
> such jobs are part of the release-informing dashboard for 1.15:
> https://k8s-testgrid.appspot.com/sig-release-1.15-informing
> 
> ```
> kubeadm-kinder-upgrade-1-14-1-15
> kubeadm-kinder-1-15-on-1-14
> ```
> 
> with the CI artifacts for 1.14 missing (or corrupt as explain [here](https://github.com/kubernetes/kubernetes/issues/86182#issuecomment-565818803)) these 1.15 jobs will remain red.
> 
> EDIT: i guess, what i'm trying to say is that even if we remove the 1.14 jobs, the artifacts are still corrupt.
> xref [kubernetes/test-infra#15875](https://github.com/kubernetes/test-infra/pull/15875)

w.r.t.
> @justaugustus why are 1.14 jobs being removed at this time?

Discussed this with the Release Team and Branch Managers today at the Release Team meeting.
We need to disambiguate the branch cut process. There are two steps: removing the old branch jobs, generating new branch jobs (and rotating configs).

The latter can only happen after a new release branch has been created.
Removing the oldest jobs, however, can and should happen earlier in the cycle. This means that we get to turn off CI for non-supported releases sooner rather than later, which is good for $$$. Right now, I'm thinking the "right" time to do this is as we cut the first patch release for the previous minor version (e.g., 1.17.1 is being cut tomorrow).

This PR is a step in that new direction.

**Comment by dchen1107** [comment]
I looked at two commits related to SIG-node: 

sig-node: Remove sig-node-cri-1.12 dashboard …
sig-node: Fix testgrid annotations and suffixes for node-kubelet jobs …

/lgtm

**Comment by justaugustus** [comment]
Thanks for the review, @dchen1107!

File: releng/OWNERS [hunk 1]
```diff
@@ -3,6 +3,7 @@
 approvers:
 - calebamiles
 - justaugustus
+- Katharine
```
Comment (BenTheElder) [inline releng/OWNERS]: /assign @Katharine

Comment (justaugustus) [reply releng/OWNERS]: (already assigned :) )

Comment (BenTheElder) [reply releng/OWNERS]: stephen you're ruining my gubernator 0 attempt :upside_down_face:

Comment (justaugustus) [reply releng/OWNERS]: Oh yeah? :stuck_out_tongue_winking_eye:

**Comment by BenTheElder** [comment]
kind looks about right for now, good luck @justaugustus 
/uncc

**Comment by justaugustus** [comment]
I've opened the following PRs to break out some of the work that was happening here:

- https://github.com/kubernetes/test-infra/pull/15920
  4ae62b0c37354df805d75f359ad31d3d01d63d5a OWNERS: Add Katharine and justaugustus as 
  approvers for generated tests

- https://github.com/kubernetes/test-infra/pull/15921
  dcf62e2a3ec7310a2ef582789769c4fb7f3358ff sig-node: Remove sig-node-cri-1.12 dashboard
  b00d8590c039878084f26fe51686be5aafc41eb2 sig-node: Fix testgrid annotations and suffixes for 
  node-kubelet jobs

- https://github.com/kubernetes/test-infra/pull/15922
  c0bf75bf1b5899bdbf962660dc6a434a9594525a provider/gcp: Update dashboard tabs to reference 
  correct pseudo-versions

- https://github.com/kubernetes/test-infra/pull/15923
  kind removals from f4b2d000cb7cde0e5ac570c19861301f01a3b3a1
  000308ac64e796067adfff631b210bd77371069b kind: Add 1.17 release-blocking tests

I'll rebase those commits out and update the description once the other PRs merge.

**Comment by justaugustus** [comment]
All of the commits that I've peeled off of this PR have merged as individual PRs.
Once we complete the patch release set (1.17 - 1.14, ref: https://github.com/kubernetes/kubernetes/issues/86182) next week, we'll be able to merge this and formally turn off tests for 1.14.

cc: @tpepper 
/unassign @aojea @dchen1107

**Comment by justaugustus** [comment]
(This should merge after https://github.com/kubernetes/test-infra/pull/15564.)

**Comment by justaugustus** [comment]
Rebased now that https://github.com/kubernetes/test-infra/pull/15564 is merged.
This is ready for approval.

/hold cancel

**Comment by justaugustus** [comment]
Aha, [my comment on another PR](https://github.com/kubernetes/test-infra/pull/15110#issuecomment-554687970) rears its' head again:

> (Looks like some additional tests were enabled recently to enforce OWNERS and descriptions for certain job types. Some of those are my jobs, so I'll fix some of that up.)

I'm not sure why this hasn't failed on other PRs, but I can work on fixing it up here.

**Comment by justaugustus** [comment]
Given that the branch cut is around the corner and I'd like the Branch Managers (@cpanato, @saschagrunert) to get familiar with the process, we'll continue as-is and I'll close this out.

/close

**Pull request closed without merging**